### PR TITLE
Ruby 3 compatibility

### PR DESCRIPTION
**Pull request by nadamsoreilly**
This PR updates the Faraday version so that this gem is compatible with Ruby 3. It also did some housekeeping and replaced the Travis CI with GitHub actions CI.

I'm not sure why, but I couldn't add either the PubEng or Tools team as reviewers. I'm going to send a separate Solutions ticket to be made and Admin so I can manage the permissions.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    atlas-api (0.1.2)
+    atlas-api (1.0.0)
```
Comment (ruralocity) [inline Gemfile.lock:4]: Possibly overthinking things, but moving from a 0.x to a 1.x conveys API stability that we may not want to guarantee. I'm not 100% sure if bumping the major version is necessary if we're dropping language version support, but not making changes to the API. We might want to ask around in our engineering channels for guidance.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - master
+
+jobs:
+  tests:
+    name: Run test suite
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        ruby-version: ['3.0', '2.7']
```
Comment (ruralocity) [inline .github/workflows/ci.yaml:18]: Love this! Should we add 3.1 and 3.2 as well?

**Review by ruralocity — COMMENTED** [review]
Couple of suggestions/questions. Overall this looks great, glad to see us heading in this direction!

File: Gemfile.lock (hunk 1 above)
Comment (nadamsoreilly) [reply Gemfile.lock:4]: That's a good point. I was thinking of it in terms of the Tools team taking ownership of active development, but I agree that we haven't made API changes or guaranteed stability there. I'll drop back to 0.2.0.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (nadamsoreilly) [reply .github/workflows/ci.yaml]: Just added them and it works!

File: atlas-api.gemspec [hunk 3]
```diff
@@ -17,7 +17,7 @@ Gem::Specification.new do |gem|
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.require_paths = ["lib"]
 
-  gem.add_dependency "faraday", "~> 0.15.4"
+  gem.add_dependency "faraday", "~> 2.7.0"
```
Comment (delfanbaum) [inline atlas-api.gemspec:20]: Wow that's quite a version jump....

**Review by delfanbaum — APPROVED** [review]
Great work!

**Pull request merged**